subxt/src/client/lightclient/background.rs

```diff
@@ -0,0 +1,439 @@
+// Copyright 2019-2023 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+use futures::stream::StreamExt;
+use futures_util::future::{self, Either};
+use serde::Deserialize;
+use serde_json::value::RawValue;
+use std::{collections::HashMap, str::FromStr, sync::Arc};
+use tokio::sync::{mpsc, oneshot};
+
+use super::LightClientError;
+use smoldot_light::{platform::default::DefaultPlatform as Platform, ChainId};
+
+const LOG_TARGET: &str = "light-client-background";
+
+/// The response of an RPC method.
+pub type MethodResponse = Result<Box<RawValue>, LightClientError>;
+
+/// The type of a subscription ID.
+type SubscriptionId = usize;
```

Comment:
nit: I would probably do the same for `RequestId`, that are also just represented as a usize or remove the type alias.